stm32-modules/heater-shaker/firmware/system/system_hardware.c

```diff
@@ -60,3 +90,149 @@ asm volatile (
   : "r" (*sysmem_boot_loc)
   : "memory"  );
 }
+
+bool system_hardware_setup_led(void) {
+  uint32_t ulNotificationValue;
+  const TickType_t xMaxBlockTime = pdMS_TO_TICKS( 100 );
+  HAL_StatusTypeDef status;
+
+  configASSERT( xTaskToNotify == NULL );
```

Comment:
are `configAsserts` the right way to handle this? should we instead be returning early and sending an error?